### PR TITLE
[Merged by Bors] - feat(CategoryTheory): morphisms of localizers

### PR DESCRIPTION
**Pull request by joelriou**
This PR introduces the categorical notion of morphisms of localizers.

---
- [x] depends on: #6867 

This notion will be used in order to prove results about the composition of localization functors. (In a more remote future, it will be useful also in the study of derived functors: I intend to formalize some results from the paper _Structures de dérivabilité_ by Bruno Kahn and Georges Maltsiniotis which is given as a reference here.)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Localization/LocalizerMorphism.lean [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+/-!
+# Morphisms of localizers
+
+A morphism of localizers consists of a functor `F : C₁ ⥤ C₂` between
+two categories equipped with morphism properties `W₁` and `W₂` such
+that `F` sends morphisms in `W₁` to morphisms in `W₂`.
+
+If `Φ : LocalizerMorphism W₁ W₂`, we introduce the condition
+`Φ.IsLocalizedEquivalence` which expresses that the induced functor
+on the localized categories is a equivalence. This condition is
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Localization/LocalizerMorphism.lean:17]:
```suggestion
on the localized categories is an equivalence. This condition is
```

File: Mathlib/CategoryTheory/Localization/LocalizerMorphism.lean [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+variable (Φ : LocalizerMorphism W₁ W₂) (L₁ : C₁ ⥤ D₁) [L₁.IsLocalization W₁]
+  (L₂ : C₂ ⥤ D₂) [L₂.IsLocalization W₂]
+
+lemma inverts : W₁.IsInvertedBy (Φ.functor ⋙ L₂) :=
+  fun _ _ _ hf => Localization.inverts L₂ W₂ _ (Φ.map _ hf)
+
+/-- When `Φ : LocalizerMorphism W₁ W₂` and that `L₁` and `L₂` are localization functors
+for `W₁` and `W₂`, then `Φ.localizedFunctor L₁ L₂` is the induced functor on the
+localized categories. --/
+noncomputable def localizedFunctor : D₁ ⥤ D₂ :=
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Localization/LocalizerMorphism.lean:71]: Should this def be mentioned in the module docstring at the top of the file?

Comment (joelriou) [reply Mathlib/CategoryTheory/Localization/LocalizerMorphism.lean:73]: Thanks for the suggestion!

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**